### DISCUSSION TITLE
How to add pivot field without use repeatable?

### DISCUSSION THREAD

**Discussion by cod3rshotout** [Laravel-Backpack/community-forum#1198] [Q&A (Help)] [answered]
Hello, I have created a field that read the content from a pivot table:

    [
      'name' => 'permissionPages',
      'tab' => $permissions_tab,
      ]
    ]

This is the relationship:

    public function permissionPages()
    {
        return $this->belongsToMany(
            Page::class,
            'user_permission_contents',
            'content_id',
            'id',
        );
    }

Basically I need to pass the "user_id" to the pivot table called "user_permission_contents" which have the following columns: content_id, content_type, user_id.
Now the problem's that if I specify the field as subfield:

      'subfields' => [
        [
          'name' => 'user_id'
        ]

I also get a repeatable structure, but I just need a select 2 with a list of Page. How can I pass the user_id field to the pivot table?

**Comment by jcastroa87** [comment, marked as answer]
To answer your question about passing the `user_id` to the pivot table while maintaining a select2 list of Pages, we can use the `relationship` field type with some specific configurations. Here's how you can achieve this:

1. First, modify your relationship definition to include the `user_id` in the `withPivot()` method:

```php
public function permissionPages()
{
    return $this->belongsToMany(
        Page::class,
        'user_permission_contents',
        'content_id',
        'id',
    )->withPivot('user_id');
}
```

2. Then, in your CRUD controller, define the field like this:

```php
CRUD::field([
    'name' => 'permissionPages',
    'type' => 'relationship',
    'tab' => $permissions_tab,
    'subfields' => [
        [
            'name' => 'user_id',
            'type' => 'hidden',
            'value' => backpack_user()->id, // Assuming you're using Backpack's authentication
        ]
    ],
]);
```

This configuration will:
- Use the `relationship` field type, which will display a select2 list of Pages.
- Include a hidden subfield for `user_id`, which will be automatically filled with the current user's ID.

The `relationship` field type will handle saving the pivot data, including the `user_id`, without creating a repeatable structure.

For more details on how to use the `relationship` field type and save additional data to pivot tables, you can refer to the [Backpack documentation on relationship fields](https://backpackforlaravel.com/docs/6.x/crud-fields#relationship-pro).

If you need to customize the options shown in the select2 list or add any other configurations, you can add them to the field definition. For example:

```php
CRUD::field([
    'name' => 'permissionPages',
    'type' => 'relationship',
    'tab' => $permissions_tab,
    'attribute' => 'page_title', // Assuming the Page model has a 'page_title' attribute
    'placeholder' => 'Select pages',
    'subfields' => [
        [
            'name' => 'user_id',
            'type' => 'hidden',
            'value' => backpack_user()->id,
        ]
    ],
]);
```

This approach should allow you to pass the `user_id` to the pivot table while maintaining a simple select2 interface for choosing Pages.
